js/Common.js

```diff
@@ -1638,8 +1648,28 @@ if (!CRM.vars) CRM.vars = {};
 
   // Format a date for output to the user
   // Input may be a js Date object, a unix timestamp or a yyyy-mm-dd string
-  CRM.utils.formatDate = function(input, outputFormat) {
-    return input ? $.datepicker.formatDate(outputFormat || CRM.config.dateInputFormat, CRM.utils.makeDate(input)) : '';
+  CRM.utils.formatDate = function(input, outputFormat, showTime) {
+    if (!input) {
+      return '';
+    }
+    var date = CRM.utils.makeDate(input),
+      output = $.datepicker.formatDate(outputFormat || CRM.config.dateInputFormat, date);
+    if (showTime) {
+      var hour = date.getHours(),
+        min = date.getMinutes(),
+        suf = '';
+      if (showTime === 12 || (showTime === true && !CRM.config.timeIs24Hr)) {
+        suf = ' ' + ts('AM');
```

Comment:
showTime can be the number 12 or a boolean?

Reply:
Also might consider using the same or similar code for this whole block that the existing timeentry plugin uses, something like `this._formatNumber(inst.options.show24Hours ? inst._selectedHour : ((inst._selectedHour + 11) % 12) + 1)`.

Reply:
@demeritcowboy thanks for the review.
The `showTime` param can be either `12`, `24`, or `true`. If the latter, the system default is used.

I agree that this code basically duplicates some of the timeentry plugin, but this is only a few lines long and I'd rather have a little redundant code than rely on calling a private function in a plugin we don't control.